src/dicomweb_client/uri.py

```diff
@@ -0,0 +1,358 @@
+            Study Instance UID to use in the new `URI` or `None` if the
+            `study_instance_uid` from the current `URI` should be used.
+        series_instance_uid: str, optional
+            Series Instance UID to use in the new `URI` or `None` if the
+            `series_instance_uid` from the current `URI` should be used.
+        sop_instance_uid: str, optional
+            SOP Instance UID to use in the new `URI` or `None` if the
+            `sop_instance_uid` from the current `URI` should be used.
+
+        Returns
+        -------
+        URI
+          The newly constructed `URI` object.
+
+        Raises
+        ------
+        ValueError
+            If the new `URI` is invalid (e.g., if only the SOP Instance UID is
+            specified, but the Series Instance UID is missing in the current
+            `URI`).
+        """
+        return URI(
+            base_url if base_url is not None else self.base_url,
+            study_instance_uid
+            if study_instance_uid is not None else self.study_instance_uid,
+            series_instance_uid
+            if series_instance_uid is not None else self.series_instance_uid,
+            sop_instance_uid
+            if sop_instance_uid is not None else self.sop_instance_uid,
+        )
+
+    @property
+    def parent(self) -> 'URI':
+        """Returns a URI to the "parent" resource.
+
+        Depending on the `type` of the current `URI`, the `URI` of the parent
+        resource is defined as:
+
+        +----------+---------+
+        | Current  | Parent  |
+        +==========+=========+
+        | Service  | Service |
+        +----------+---------+
+        | Study    | Service |
+        +----------+---------+
+        | Series   | Study   |
+        +----------+---------+
+        | Instance | Series  |
+        +----------+---------+
+
+        Returns
+        -------
+        URI
+          An instance pointing to the parent resource.
+        """
+        if self.type == URIType.SERVICE:
+            return self
+        elif self.type == URIType.STUDY:
+          return URI(self.base_url)
+        elif self.type == URIType.SERIES:
+            return self.study_uri()
+        else:
+            return self.series_uri()
+
+    @property
+    def parts(self) -> Tuple[str, ...]:
+        """Returns the sequence of URI components in a `tuple`.
+
+        For example, if the URI is:
+        ``https://service.com/studies/1.2.3/series/4.5.6``
+
+        then the method returns ``['https://service.com', '1.2.3', '4.5.6']``
+
+        Returns
+        -------
+        Tuple[str, ...]
+            Sequence of URI components.
+        """
+        return tuple(part for part in (self.base_url, self.study_instance_uid,
+                                       self.series_instance_uid,
+                                       self.sop_instance_uid)
+                     if part is not None)
+
+    @classmethod
+    def from_string(cls,
```

Comment:
We may want to allow suffixes such as `/metadata` and store them on the object.

Reply:
Done.